model/src/test/java/org/projectnessie/model/TestNamespace.java

```diff
@@ -85,19 +85,19 @@ public void testOneElement() {
   public void testRoundTrip() {
     List<String> elements = asList("a", "b.c", "namespace");
     String pathString = UriUtil.toPathString(elements);
-    String expectedPathString = "a.b\u0000c.namespace";
+    String expectedPathString = "a.b%00c.namespace";
     assertThat(pathString).isEqualTo(expectedPathString);
     Namespace namespace = Namespace.parse(pathString);
     assertThat(namespace.name()).isEqualTo(pathString);
     assertThat(namespace.getElements()).isEqualTo(elements);
     assertThat(namespace.toString()).isEqualTo(pathString);
     assertThat(namespace.toPathString()).isEqualTo(pathString);
     assertThat(namespace.name()).startsWith("a.b");
-    assertThat(namespace.name()).startsWith("a.b\u0000");
-    assertThat(namespace.name()).startsWith("a.b\u0000c");
-    assertThat(namespace.name()).startsWith("a.b\u0000c.namespa");
+    assertThat(namespace.name()).startsWith("a.b%00");
```

Comment:
Not happy with replacing `\u0000` with `%00` - those are different strings. Better keep `\u0000` for `ContentKey` and keep URL encoding exclusively to the REST layer

Reply:
My understanding is that Jetty does not permit null chars in paths. So we cannot use the null char is parameters that come from URL paths. @nastra probably has more knowledge on this, though.

Reply:
Right - but `ContentKey` gets constructed from a URL-encoded string (with`%00`).
So `%00` is the representation in URLs and `\u0000` is the representation in `ContentKey`.
IMO it should be sufficient to handle `%00` in `UriUtil` via `ContentKeyParamConverterProvider` & `NamespaceParamConverterProvider`

Reply:
see my comment above about having one canonical representation.